### ISSUE TITLE
Bessel beams

### ISSUE THREAD

**Issue by GoogleCodeExporter** [adda-team/adda#209]
```
It should be easy to implement Bessel beams in ADDA. They are becoming more and 
more popular, so applications should come by.
```

Original issue reported on code.google.com by `yurkin` on 2 Jul 2015 at 8:59

**Comment by myurkin** [comment]
Proof-of-principle is available at https://github.com/stefanyagl/adda

**Comment by stefaniagl** [comment]
6 Bessel beam types have been implemented - LE - linear electric field, LM - linear magnetic field, CS - circularly symmetric energy density, TEC, TMC - polarizations generating TE and TM Bessel beams, and M - generalized type.

**Comment by myurkin** [comment]
After the pull request (#304) will be merged we still need the following to finalize this issue (to be submitted as the new pull request):
- new tests in `tests/2exec` (new command lines in suites, and potentially ignores for comparison with previous versions).
- new tests in `tests/2equiv` (specification of M matrix vs. simple types).
- description of new beams in the manual (maybe even mention some formulae, but in a concise manner)

**Comment by myurkin** [comment]
Non-trivial application of Bessel beams for imitating circular polarization of a plane wave are discussed at https://groups.google.com/g/adda-discuss/c/k-ClVwvwQQk . In particular, interesting combinations are used in [tests/equiv/ext_CD.py](../blob/master/tests/equiv/ext_CD.py). One of them highlights the lack of robustness for M matrices corresponding to TE and TM beams for very small half-cone angles.
